python/pyspark/sql/functions.py

```diff
@@ -1315,9 +1550,16 @@ def date_format(date, format):
     pattern letters of `datetime pattern`_. can be used.
 
     .. _datetime pattern: https://spark.apache.org/docs/latest/sql-ref-datetime-pattern.html
-    .. note:: Use when ever possible specialized functions like `year`. These benefit from a
-        specialized implementation.
 
+    .. versionadded:: 1.5.0
+
+    Notes
+    -----
+    Use when ever possible specialized functions like `year`. These benefit from a
```

Comment:
I know this hasn't been added in this PR, but the grammar seems off. Maybe we could change it to something like `Whenever possible, use  specialized functions like `year``?